Change some sample disapprove comments to feel less 'stern'. (#1264)
exercism/website-copy: automated-comments/go/two-fer/strings_join_used_for_concatenation.md

> `strings.Join` is only normally used if the data is of type `[]string`.

Reviewer: See comment below about being Dutch, but I like the below:

```suggestion
`strings.Join` is usually only used if the data is of type `[]string`.
```

Reviewer 2: It is "more common" than what?  Perhaps "it is common" is better here.

> In other situations, it is normal to use `fmt.Sprintf`, which is a very powerful way of building up strings from multiple pieces.

Reviewer: Maybe this is me being Dutch, but "normal" felt a bit odd to me. It felt a bit like saying that not doing it is abnormal. Maybe "customary" is better? Feel free to ignore if this really _is_ me being Dutch :D

```suggestion
In other situations, it is customary to use `fmt.Sprintf`, which is a very powerful way of building up strings from multiple pieces.
```

Reviewer 3: 'common' ? 'more common' ?

Author: It depends on whether it's normal or common! :) If nearly everyone does it, then I'd say normal (e.g. it's an idiom or close to it). If it's common (eg. 60% of people do it) then I'd say "common" but then is it in an idiom and if not, should we not say it's common or normal, but just state it as an alternative?

Reviewer: Thanks for the explanation. I'll then leave it to the go-people to determine if it is normal or common :)

Reviewer 4: It's not like everybody does it, no. So `common` seems correct according to the definition @iHiD gave.

Reviewer 5: Perhaps "more common" is the best compromise here. I often like to say that certain things in Go are "conventional", too—and I mean that in a good way! Gophers love being conventional.

Author: ```suggestion
In other situations, it is more common to use `fmt.Sprintf`, which is a very powerful way of building up strings from multiple pieces.
```

Reviewer 2: It is "more common" than what? Perhaps "it is common" is better here.

> In other situations, it is normal to use `fmt.Sprintf`, which is a very powerful way of building up strings from multiple pieces.

Reviewer 4: ```suggestion
In other situations, it is common to use `fmt.Sprintf`, which is a very powerful way of building up strings from multiple pieces.
```

Reviewer 4: I'd go with `common`.

> Could you try updating your solution to use that?

Reviewer: Maybe be explicit about what to use here:

```suggestion
Could you try updating your solution to use `fmt.Sprintf`?
```

@kotp In other comments, you've suggested to remove the "you" part, which I like. Do you have any suggestion how to apply it here?

Reviewer 3: I feel this is a different 'you', and totally okay to use here.

Reviewer 5: I often phrase this as "Would you like to try...?" That makes it clear that this isn't a _correction_, as such; it's an alternative suggestion which may be worth exploring.

(Of course, the answer may be "No, I wouldn't". That's never happened yet, though.)

Author: > I feel this is a different 'you', and totally okay to use here.

Yes! Agreed.

> Would you like to try..

Agree too.

```suggestion
Would you like to try updating your solution to use that?
```

Reviewer 6: > (Of course, the answer may be "No, I wouldn't". That's never happened yet, though.)

On the JS track we get this _quite frequently_, which is why I stopped phrasing it like that.

Reviewer 5: > How does using that improve the code?

This is a perfectly sensible and nice suggestion. However, students tend to take such questions a bit too seriously, and I don't like the idea of someone thinking too hard about this (remember, this is the very first program they've ever written in Go).

We've talked a lot about the string concatenation issue and how much information and nuance it's wise to load students with on this first exercise. I favour a light touch here; if they use `fmt.Sprintf`, great. If not, I'll let them know that it exists and it's a powerful way to format data into strings. If they're not particularly interested now, they'll learn about it soon enough.

I'd ask if they'd like to try updating their solution to use `fmt.Sprintf`, and leave it at that.

Reviewer 5: For comparison, here's what Exalysis says if the student uses `+=` to concatenate strings. As usual, @tehsphinx strikes a very warm and genial note:

> I see you are using `+` to concatenate the string. That is perfectly legit! If you didn't know about the `fmt` package, check it out. It can do lots of things related to string formatting, including concatenating strings.
>
> (Using `+` is probably the right choice here. I just wanted you to know about `fmt`, because it's awesome, and you'll be using it a lot.)

Reviewer 4: I don't think that was entirely me...

Reviewer: > This is a perfectly sensible and nice suggestion. However, students tend to take such questions a bit too seriously, and I don't like the idea of someone thinking too hard about this

I feel the same way.

Author: I feel like being explicit that we expect an action is important as I too have had lots of people say "no" when I phrase things as "would you like to.." and have had essays rather than code back when I ask questions that don't explicitly tell people to submit code.

How about we make it more explicit that we want/expect someone to do something, but still polite (I'm english - it's what we do :))

```suggestion
Could you try updating your solution to use `fmt.Sprintf` please?
```

> Could you try updating your solution to use that?

Reviewer 4: ```suggestion
Can you try updating your solution to use `fmt.Sprintf`?
```

Reviewer 4: `Would you like to try...` feels too soft. We are talking about a `disapprove` comment here. I'd even suggest to use `Can` if we want to make it a question.

Reviewer 4: > `How does using that improve the code?`

We have such a question in another comment and have lots of students actually answer that. If we want answers, that is a good thing but here I would not try to get actual answers.

Reviewer 5: > We are talking about a disapprove comment here.

I think this discussion is gradually bringing us to the conclusion that this _shouldn't_ be a disapprove. It's a 'would you like to try'.

Reviewer 4: If that is the case, we have to make it `stern` again because it will not be copied in by a mentor but provided automatically by system. Asking a question would then not be a good idea at all.

Reviewer 5: "Only an idiot would use `strings.Join()` in this situation. Therefore you must be an idiot. Please get a clue and stop wasting my time."

Author: I'm not sure that's neutral "bot" tone enough for an approve comment, @bitfield :)

I think it would be helpful to get away from the word "stern" as that's not the intention. The intention for approval comments is that they're not **personal** - ie they don't feel like they're coming from a person. They should still be empathetic but they should be factual, rather than being a basis for discussion or conversation.

Reviewer 5: This has never really made sense to me. In my experience people learn better, and respond better, when the comments you give them are friendly, encouraging, and, yes, personal.

Sure, they're _selected_ by machine, but they're written by humans. Why shouldn't they sound human?
